demos/python_demos/image_translation_demo/image_translation_demo.py

```diff
@@ -0,0 +1,135 @@
+                      help="Optional. Path to an .xml file with a trained semantic segmentation model",
+                      default=None, type=str)
+    args.add_argument("-ii", "--input_images",
+                      help="Optional. Path to a folder with input images or path to a input image",
+                      type=str)
+    args.add_argument("-is", "--input_semantics",
+                      help="Optional. Path to a folder with semantic images or path to a semantic image",
+                      type=str)
+    args.add_argument("-ri", "--reference_images",
+                      help="Required. Path to a folder with reference images or path to a reference image",
+                      required=True, type=str)
+    args.add_argument("-rs", "--reference_semantics",
+                      help="Optional. Path to a folder with reference semantics or path to a reference semantic",
+                      type=str)
+    args.add_argument("-o", "--output_dir", help="Required. Path to a folder where output files will be saved",
+                      required=True, type=str)
+    args.add_argument("-d", "--device",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL or MYRIAD is "
+                           "acceptable. Default value is CPU",
+                      default="CPU", type=str)
+    return parser
+
+
+def get_files(path):
+    if path is None:
+        return []
+    if os.path.isdir(path):
+        file_paths = [os.path.join(path, file) for file in os.listdir(path) if os.path.isfile(os.path.join(path, file))]
+        return sorted(file_paths)
+    return [path]
+
+
+def get_mask_from_image(image, model):
+    image = preprocess_for_seg_model(image, input_size=model.input_size)
+    res = model.infer(image)
+    mask = np.argmax(res, axis=1)
+    mask = np.squeeze(mask, 0)
+    return mask + 1
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    log.info("Creating CoCosNet Model")
+    ie_core = IECore()
+
+    gan_model = CocosnetModel(ie_core, args.translation_model,
+                              args.translation_model.replace(".xml", ".bin"),
+                              args.device)
+    seg_model = SegmentationModel(ie_core, args.segmentation_model,
+                                  args.segmentation_model.replace(".xml", ".bin"),
+                                  args.device) if args.segmentation_model else None
+
+    log.info("Preparing input data")
+    input_data = []
+    assert args.reference_images and ((args.input_semantics and args.reference_semantics)
+                                      or (args.input_images and args.segmentation_model)
+                                      ), "Not enough data to do inference"
+    use_seg = not (args.reference_images and args.input_semantics and args.reference_semantics)
```

Comment:
fixed, but using | instead ^ in assert. If you will use ^, you get error "Don't know where to get data" when all parameters are set

Reply:
That's the point. If everything is set you will just abandon two of args. Instead of error you could print a warning, but demo shouldn't silently make a guess.

Reply:
Ok, done